code_style.md

```diff
@@ -0,0 +1,455 @@
+# Matrix Code Style Guide
```

Comment:
Should this be titled "Element Web Style Guide" given that it's specifically for the element-web codebase?

Reply:
Copied the description from the code style PR